src/main/java/edu/ie3/datamodel/io/source/csv/CsvIdCoordinateSource.java

```diff
@@ -181,23 +187,26 @@ protected Stream<Map<String, String>> buildStreamWithFieldsToAttributesMap()
 
       Function<Map<String, String>, String> idExtractor =
           fieldToValues -> fieldToValues.get(factory.getIdField());
-      Set<Map<String, String>> withDistinctCoordinateId =
+      Try<Set<Map<String, String>>, SourceException> withDistinctCoordinateId =
           dataSource.distinctRowsWithLog(
               allRows, idExtractor, COORDINATE_ID_MAPPING, "coordinate id");
       Function<Map<String, String>, String> coordinateExtractor =
           fieldToValues ->
               fieldToValues
                   .get(factory.getLatField())
                   .concat(fieldToValues.get(factory.getLonField()));
-      return dataSource
-          .distinctRowsWithLog(
-              withDistinctCoordinateId, coordinateExtractor, COORDINATE_ID_MAPPING, "coordinate")
-          .parallelStream();
+
+      return withDistinctCoordinateId
+          .flatMap(
+              set ->
+                  dataSource.distinctRowsWithLog(
+                      set, coordinateExtractor, COORDINATE_ID_MAPPING, "coordinate"))
+          .map(Set::parallelStream);
     } catch (IOException e) {
       log.error("Cannot read the file for coordinate id to coordinate mapping.", e);
+      return Success.of(Stream.empty());
```

Comment:
Changing this part will result SIMONA to stop running if the file cannot be read. With the current implementation SIMONA would use an empty map and return an empty list of `CoordinateDistance` when there is a request.

Therefore changing this part is a good idea. But we must always provide a valide file to run SIMONA with a weather source.

Reply:
Ok but is it then still possible to run SIMONA without weather?
It should work like this: 
- weather service enabled by configuration -> weather data and coordinate data required
- weather service disabled by configuration -> we do not even check if weather/coordinate data is available because we don't care

Reply:
With the current implementation in SIMONA we always have to provide a weather service (either CSV, SQL or Sample).

By using the sample configuration there is no problem with the PSDM changes. The changes in this PR are only relevant if a csv source is used in SIMONA.

Also there is currently no "proper" way to disable weather services in SIMONA, only the workaround to configure a sample source. This behavior could be changed in (https://github.com/ie3-institute/simona/issues/180).